ipalib/util.py

```diff
@@ -1078,8 +1077,8 @@ def check_client_configuration():
     """
     Check if IPA client is configured on the system.
     """
-    fstore = sysrestore.FileStore(paths.IPA_CLIENT_SYSRESTORE)
-    if not fstore.has_files() and not os.path.exists(paths.IPA_DEFAULT_CONF):
+    if (not any(os.listdir(_IPA_CLIENT_SYSRESTORE)) and not
```

Comment:
This will fail with an exception if the directory does not exist:

```
>>> os.listdir('/non/existing')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '/non/existing'
```

How about:
```
   if (not os.path.isfile(_IPA_DEFAULT_CONF) or
           not os.path.isdir(_IPA_CLIENT_SYSRESTORE) or
           not os.listdir(_IPA_CLIENT_SYSRESTORE)):
```